opentelemetry-api/src/metrics/meter.rs

```diff
@@ -1,369 +1,165 @@
-    pub fn f64_value_observer<T, F>(&self, name: T, callback: F) -> ValueObserverBuilder<'_, f64>
-    where
-        T: Into<String>,
-        F: Fn(ObserverResult<f64>) + Send + Sync + 'static,
-    {
-        ValueObserverBuilder::new(
-            self,
-            name.into(),
-            Some(AsyncRunner::F64(Box::new(callback))),
-            NumberKind::F64,
-        )
+    /// creates an instrument for recording a distribution of values.
+    pub fn u64_histogram(&self, name: impl Into<String>) -> InstrumentBuilder<'_, Histogram<u64>> {
+        InstrumentBuilder::new(self, name.into())
     }
 
-    /// Creates a new `BatchObserver` that supports making batches of observations
-    /// for multiple instruments or returns an error if instrument initialization
-    /// fails.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use opentelemetry_api::{global, metrics::BatchObserverResult, KeyValue};
-    ///
-    /// # fn init_observer() -> opentelemetry_api::metrics::Result<()> {
-    /// let meter = global::meter("test");
-    ///
-    /// meter.build_batch_observer(|batch| {
-    ///   let instrument = batch.u64_value_observer("test_instrument").try_init()?;
-    ///
-    ///   Ok(move |result: BatchObserverResult| {
-    ///     result.observe(&[KeyValue::new("my-key", "my-value")], &[instrument.observation(1)]);
-    ///   })
-    /// })?;
-    /// # Ok(())
-    /// # }
-    /// ```
-    pub fn build_batch_observer<B, F>(&self, builder: B) -> Result<()>
-    where
-        B: Fn(BatchObserver<'_>) -> Result<F>,
-        F: Fn(BatchObserverResult) + Send + Sync + 'static,
-    {
-        let observer = builder(BatchObserver::new(self))?;
-        self.core
-            .new_batch_observer(AsyncRunner::Batch(Box::new(observer)))
+    /// creates an instrument for recording a distribution of values.
+    pub fn i64_histogram(&self, name: impl Into<String>) -> InstrumentBuilder<'_, Histogram<i64>> {
+        InstrumentBuilder::new(self, name.into())
     }
 
-    /// Creates a new `BatchObserver` that supports making batches of observations
-    /// for multiple instruments.
-    ///
-    /// # Panics
-    ///
-    /// Panics if instrument initialization or observer registration returns an
-    /// error.
-    ///
-    /// # Examples
+    /// Captures the function that will be called during data collection.
     ///
-    /// ```
-    /// use opentelemetry_api::{global, metrics::BatchObserverResult, KeyValue};
-    ///
-    /// let meter = global::meter("test");
-    ///
-    /// meter.batch_observer(|batch| {
-    ///   let instrument = batch.u64_value_observer("test_instrument").init();
-    ///
-    ///   move |result: BatchObserverResult| {
-    ///     result.observe(&[KeyValue::new("my-key", "my-value")], &[instrument.observation(1)]);
-    ///   }
-    /// });
-    /// ```
-    pub fn batch_observer<B, F>(&self, builder: B)
+    /// It is only valid to call `observe` within the scope of the passed function.
```

Comment:
This is kind of confusing. I suppose this is saying we cannot use Sync* instrument in callback?

Reply:
yeah that's the intention